src/Components/Matrix/Matrix.js

```diff
@@ -0,0 +1,68 @@
+import React, {useState} from 'react';
+import { useForm } from 'react-hook-form';
+import RadioItem from './RadioItem';
+import './Matrix.scss';
+
+const Matrix=({ qno, data, id, isEdit = false, onUpdate})=> { 
+    
+    const onTitleUpdate = (value) => {
+        let tempData = { ...data };
+        tempData.title = value.target.value;
+        onUpdate(id,tempData);
+    }
+    const onButtonDetailsUpdate = (value) => {
+        let tempData = {...data};
+        tempData.ans= value.target.value;
+        onUpdate(id,tempData);
+    }
+    const selectTabs = (index, value) => {
+        let tempData = {...data};
+        tempData.ansType = value;
+        onUpdate(id,tempData);
+    }
+
+    const [optionList, setOptionList] = useState([
+        {name:'Quention 1', value: ''},
```

Comment:
data should have all the details and no local variable which holds some information.

Reply:
if the matrix component will be used by multiple screens then all the data should come dynamically as data prop